flink-java/src/main/java/org/apache/flink/api/java/typeutils/TupleTypeInfo.java

```diff
@@ -174,13 +177,15 @@ public String toString() {
 			
 			TypeInformation<?> info = BasicTypeInfo.getInfoFor(type);
 			if (info == null) {
-				throw new IllegalArgumentException("Type at position " + i + " is not a basic type.");
+				try {
+					info = ValueTypeInfo.getValueTypeInfo((Class<Value>) type);
```

Comment:
Any class can implement the `Value` interface.
 We need to check if the class is on of the "BasicValue" classes, that Flink ships.